src/main/java/com/hpe/adm/octane/ideplugins/services/connection/IdePluginsOctaneHttpClient.java

```diff
@@ -129,22 +130,21 @@ public IdePluginsOctaneHttpClient(String urlDomain, ClientType clientType) {
     }
 
     @Override
-    public synchronized boolean authenticate(Authentication authentication) {
+    public synchronized boolean authenticate() {
         authenticationLock.lock();
-        lastUsedAuthentication = authentication;
 
         try {
-            if (authentication instanceof GrantTokenAuthentication) {
-                return grantTokenAuthenticate(authentication);
+            if (lastUsedAuthentication instanceof GrantTokenAuthentication) {
```

Comment:
I'm a bit confused - now that you don't assign any value to lastUsedAuthentication when first entering this method, doesn't this check throw  a NPE or other error?

Reply:
No because i use a setter now to initialize lastUsedAuthentication which i usually call before calling this method

Reply:
usually? are there any cases where a NPE can be thrown?

Reply:
The authenticate method is called inside this IdePluginsOctaneHttpClient class and from outside. When it's called from outside i use the setter and when it's called from inside there is a null check before. So no, there shouldn't be any cases where NPE can be thrown